components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1642,15 +1675,14 @@ void KeyringService::SetSelectedAccount(const std::string& address,
     }
   }
 
-  auto hardware_account_info_ptrs = GetHardwareAccountsSync();
+  auto hardware_account_info_ptrs = GetHardwareAccountsSync(keyring_id);
   for (const mojom::AccountInfoPtr& info : hardware_account_info_ptrs) {
     if (base::EqualsCaseInsensitiveASCII(info->address, address)) {
       prefs_->SetString(kBraveWalletSelectedAccount, address);
```

Comment:
We need to migrate this too to support different coin type.

Reply:
Each coin type should map to its own selected account.

Reply:
done